### ISSUE TITLE
Reconnection failure attempts to call a function that does not exist

### ISSUE THREAD

**Issue by imccoy** [hgourvest/node-firebird#205]
When the connection's socket closes and it attempts to reconnect, if that reconnection fails then it attempts to emit an `'error'` event by calling `self.emit` at  https://github.com/hgourvest/node-firebird/blob/9a07f6a1d39e6adcba08e4baafc59dc493e80839/lib/index.js#L2922 and again a few lines later. `self.db.emit` does exist at that point; `self.emit` does not. The stack trace looks like this:

```
TypeError: self.emit is not a function
    at C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\index.js:2857:34
    at doError (C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\index.js:1244:9)
    at cb (C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\index.js:3201:13)
    at doCallback (C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\index.js:1234:9)
    at C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\index.js:2937:21
    at C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\messages.js:151:25
    at search (C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\messages.js:117:13)
    at C:\B\v3\hotdoc\s\svc\node_modules\node-firebird\lib\messages.js:54:21
    at FSReqWrap.wrapper [as oncomplete] (fs.js:658:17)
```

Changing `self.emit` to `self.db.emit` on those two lines is enough to avoid this escaping as an unhandled error.

However, my preference would be for reconnections not to be handled at this level, and instead to be dealt with at the application level (or by the connection pool). The main reason for this is that this reconnect behaviour can create a situation where the db "connection" is still valid but a transaction created on that connection is not. This is demonstrated by the following node repl session:

```
> let connection; NodeFirebird.attach({ /* your connection details here */ }, (err, newConnection) => {if (err) { console.log('No connection', err) } else { connection = newConnection }})
> let t1; connection.transaction((err, transaction) => { if (err) console.log('No transaction', err); else t1 = transaction });
> connection.query('select 1 from rdb$database', (err, results) => console.log(err, results)); // this works
> t1.query('select 1 from rdb$database', (err, results) => console.log(err, results)); // this works too

> connection.connection._socket.end(); // simulate a disconnect and trigger the reconnection behaviour; I believe restarting the firebird server would have the same effect.

> connection.query('select 1 from rdb$database', (err, results) => console.log(err, results)); // this still works, now over a different socket
> t1.query('select 1 from rdb$database', (err, results) => console.log(err, results)); // but this now logs an error
```

**Comment by maratth** [comment]
@mariuz `self.emit` error is already fix by pr #190

**Comment by mariuz** [comment]
@maratth Thanks , 

@imccoy Please try that fix , I merged at the moment , I will test it later
